kernel/nfft/nfft.c

```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
 
 void X(trafo_1d)(X(plan) *ths)
 {
+  if((ths->N[0] <= ths->m) || (ths->n[0] <= 2*ths->m+2))
```

Comment:
Really? Because when I do the modifications you mentioned, `simple_test.exe` has segmentation fault.